code/game/machinery/porta_turret/portable_turret.dm

```diff
@@ -420,17 +420,26 @@ DEFINE_BITFIELD(turret_flags, list(
 		power_change()
 		SetInvisibility(INVISIBILITY_NONE, id=type)
 		spark_system.start() //creates some sparks because they look cool
+		has_cover = FALSE
 		qdel(cover) //deletes the cover - no need on keeping it there!
 
+/obj/machinery/porta_turret/atom_fix()
+	set_machine_stat(machine_stat & ~BROKEN)
+	has_cover = initial(has_cover)
+	check_should_process()
+	return ..()
+
+
 /obj/machinery/porta_turret/process()
 	//the main machinery process
-	if(cover == null && anchored) //if it has no cover and is anchored
+	if(has_cover && cover == null && anchored) //if it has no cover and is anchored
 		if(machine_stat & BROKEN) //if the turret is borked
 			qdel(cover) //delete its cover, assuming it has one. Workaround for a pesky little bug
```

Comment:
this will only result in runtimes because you are qdel'ing null.
You only get to this if statement if `cover == null`